### ISSUE TITLE
mds_es_http_connect_done: HTTP connect failed, retrying...

### ISSUE THREAD

**Issue by daviddomain** [Ellerhold/fs2es-indexer#17]
Hi there,

first of all thank you for taking the effort in making this project.
Setting up the fs2es-indexer was pretty easy thanks to your detailed readme file. The debugging section is great but
what am I struggling with is getting actual search results in the mac finder. As the title of this issue suggest, my samba
log is giving me the following error...

```
[2022/05/23 19:49:48.458466,  0] ../../source3/rpc_server/mdssvc/mdssvc_es.c:272(mds_es_http_connect_done)
  mds_es_http_connect_done: HTTP connect failed, retrying...
```

Here are a couple infos on what I am working with and what are the results of going through the debugging section of your readme file:

**Samba**
---

I have a working samba service running on Ubuntu 20.04  with version 4.13

The `--with Options:` section running `smbd -b` includes `WITH_SPOTLIGHT`, so I am assuming spotlight is supported by this samba version.
Here is what the smb.conf looks like, which most of the stuff I simply copied from the samba docs.

```
[global]
   workgroup = hh.germanwahnsinn.de
   server string = %h server (Samba, Ubuntu)
   spotlight backend = elasticsearch
   elasticsearch:address = localhost
   elasticsearch:port = 9200
   log file = /var/log/samba/log.%m
   max log size = 1000
   logging = file
   server role = standalone server
   obey pam restrictions = no
   unix password sync = no
   map to guest = bad user
   veto files = /.DS_Store/
   delete veto files = yes
   min protocol = SMB2
   ea support = yes
   vfs objects = fruit streams_xattr
   fruit:metadata = stream
   fruit:model = MacSamba
   fruit:posix_rename = yes
   fruit:veto_appledouble = no
   fruit:wipe_intentionally_left_blank_rfork = yes
   fruit:delete_empty_adfiles = yes

[Johnny-Cash]
	path = /mnt/sdb/johnny-cash
	comment = Johnny-Cash
	read only = no
	browseable = yes
	force create mode = 0660
	force directory mode = 2777
	valid users = @sambashare @david
	spotlight = yes
```

**Elasticsearch**
---

I do have correctly running elasticsearch service of version 8.2

Checking the services status gives me

```
● elasticsearch.service - Elasticsearch
     Loaded: loaded (/lib/systemd/system/elasticsearch.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2022-05-23 19:49:11 UTC; 13h ago
       Docs: https://www.elastic.co
   Main PID: 1158538 (java)
      Tasks: 81 (limit: 14254)
...
```

Here the output of running `curl -X GET "localhost:9200`

```
{
  "name" : "node-1",
  "cluster_name" : "wahnsinn-intern",
  "cluster_uuid" : "FCRHucbARvuIK0SP4_8Epg",
  "version" : {
    "number" : "8.2.0",
    "build_flavor" : "default",
    "build_type" : "deb",
    "build_hash" : "b174af62e8dd9f4ac4d25875e9381ffe2b9282c5",
    "build_date" : "2022-04-20T10:35:10.180408517Z",
    "build_snapshot" : false,
    "lucene_version" : "9.1.0",
    "minimum_wire_compatibility_version" : "7.17.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "You Know, for Search"
}
```

**fs2es-indexer**
---

The tool correctly indexed a bunch of files from the configured directory.
running `fs2es-indexer search --search-path /path/to/my/directory` will give me the first 100 elasticsearch documents as expected.
Searching by using `--search-term` also gives me a bunch of results.

**Mac**
---

If I check the indexing status of the mounted samba share mounted at `/Volumes/my-share` I am getting
```
/System/Volumes/Data/Volumes/Johnny-Cash:
	Server search enabled.
```
Using `mdfind -onlyin /Volumes/my-share <search-term>` does not give any results. Not even after waiting 5 sec.
As mentioned above, checking the samba log is giving me the debug error message from the `mdssvc_es.c` file.
To make sure that I do not have any left over `.DS_Store` files and `DOSATTRIBUTES` I also deleted them as proposed.
Still no luck. Do you have any idea what could cause the issue? Would appreciate any suggestion.
Thanks in advance.

**Comment by daviddomain** [comment]
My bad. I just had to change the `elasticsearch.address` in the smb.conf and the `network.host` address in the elasticsearch.yml files. Now it is working perfectly.

**Comment by MatthiasKuehneEllerhold** [comment]
Nice :)

**Comment by TheRealJayjayli** [comment]
Hi @daviddomain, what did you updated `elasticsearch.address` and `network.host` to? I appear to have the same issue.

**Comment by daviddomain** [comment]
> Hi @daviddomain, what did you update `elasticsearch.address` and `network.host` to? I appear to have the same issue.

Hi there, not sure if I remember that correctly since I ended up using a totally different solution, but I think that I simply used the actual IP address of the host and not "localhost". I also noticed, that the fs2es-indexer did not index files with spaces or special characters in the file, or directory name. For the files that were indexed correctly it worked great though.
